src/braket/circuits/composite_operators.py

```diff
@@ -72,9 +72,17 @@ class QFT(CompositeOperator):
 
     Args:
         qubit_count (int): Number of target qubits.
+        method (bool): Optional boolean flag for decomposition,
+                         with recursive approach by default (default is True),
+                         or non-recursive approach.
     """
 
-    def __init__(self, qubit_count):
+    def __init__(self, qubit_count, method="default"):
+
+        if method != "default" and method != "recursive":
+            raise NameError("method must either be 'default' or 'recursive'.")
```

Comment:
[NameError](https://docs.python.org/3.9/library/exceptions.html#NameError) doesn't really fit this use case. This feels more like a [TypeError](https://docs.python.org/3.9/library/exceptions.html#ValueError).